x/migration/types/morse_validation.go

```diff
@@ -0,0 +1,70 @@
+package types
+
+import (
+	"crypto/ed25519"
+
+	cmted25519 "github.com/cometbft/cometbft/crypto/ed25519"
+	cosmostypes "github.com/cosmos/cosmos-sdk/types"
+)
+
+// morseClaimMessage is an interface that all Morse account/actor claim messages
+// implement which allows for a non-generic implementations of common behavior.
+type morseClaimMessage interface {
+	cosmostypes.Msg
+
+	getSigningBytes() ([]byte, error)
+
+	GetMorsePublicKey() ed25519.PublicKey
+	GetMorseSrcAddress() string
+	GetMorseSignature() []byte
+	ValidateMorseAddress() error
+	ValidateMorseSignature() error
+}
+
+// validateMorseAddress validates that the morseSrcAddress matches
+// the Morse public key of the given Morse claim message.
+func validateMorseAddress(msg morseClaimMessage) error {
```

Comment:
I left a comment in `proto/pocket/migration/tx.proto`, but don't fully understand why this is necessary.

Reply:
Given, https://github.com/pokt-network/poktroll/pull/1179#discussion_r2039511580, we need to validate this.